File: CS50/pset6/credit/credit.py
from cs50 import get_int
from sys import exit


def check(n, c):
    # Times every second number by 2
    for i in range(1, c, 2):
        n[i] = n[i] * 2
    total = 0
    temp = 0
    # Add the numbers together
    for i in range(c):
        temp = (n[i] // 10) + (n[i] % 10)
        total = total + temp
    # Retun true if it is divisible by 10
    if round(total) % 10 == 0:
        return True
    else:
        return False

# Ask for number with "do-while" loop


def main():
    while True:
        cardNumber = get_int("Number: ")
        if cardNumber > 0:
            break
    # Count the digits, while loop or len(str())?
    count = len(str(cardNumber))
    # If the card is valid
    if count != 13 and count != 15 and count != 16:
        print("INVALID")
        exit(0)
    number = list(map(int, str(cardNumber)))
    original = number.copy()
    number = list(reversed(number))
    v = check(n=number, c=count)
    # Visa test
    if (count == 13 or count == 16) and original[0] == 4 and v:
        print("VISA")
    # Amex test
    elif count == 15 and original[0] == 3 and v and (original[1] == 4 or 7):
        print("AMEX")
    # Mastercard test
    elif count == 16 and original[0] == 5 and v and original[1] in range(1, 5):
        print("MASTERCARD")
    else:
        print("INVALID")
    exit(0)


main()

File: pset1/credit/credit.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int main(void)
{
    // Declare and initialize a variable and ask for user input.
    long cardnumber = 0;

    // Ask for credit card number as long as it is above 0

    do
    {
      cardnumber = get_long("Number: ");
    }
    while (cardnumber < 0);

    // Check digit count to se if it is valid
    int count = 0;
    long long digits = cardnumber;
    while (digits > 0)
    {
        digits = digits/10;
        count++;
    }
    //if it isn't the right count it will print invalid
    if ((count != 13) && (count != 15) && (count != 16))
    {
        printf("INVALID\n");
    }

    int number[count];
    //separates each number into an array
    for (int i = 0; i < count; i++)
    {
        number[i] = cardnumber % 10;
        cardnumber = cardnumber / 10;
    }
    //creates another array that can be changed while still keeping the old one
    int originalnumber[count];
    for (int i = 1; i < count; i++)
    {
        originalnumber[i] = number[i];
    }
    //Step 1: times every second number by 2
    for (int i = 1; i < count; i+=2)
    {
        number[i] = number[i] * 2;
    }

    int v = 0;
    int temp;

    if (count == 13)
    {
      //add all the multiplied numbers together and add then to the rest of the numbers
      for (int i = 0; i < count; i++)
      {
        temp = (number[i] % 10) + (number[i] / 10 % 10);
        v = v + temp;
      }
      if (originalnumber[12] == 4 && v % 10 == 0)
      {
        printf("VISA\n");
      }
      else
      {
        printf("INVALID\n");
      }
    }
    if (count == 15)
    {
      //add all the multiplied numbers together and add then to the rest of the numbers]
      for (int i = 0; i < count; i++)
      {
        temp = (number[i] % 10) + (number[i] / 10 % 10);
        v = v + temp;
      }
    if (originalnumber[14] == 3 && v % 10 == 0 && (originalnumber[13] == 4 || originalnumber[13] == 7))
  {
    printf("AMEX\n");
  }
     else
    {
      printf("INVALID\n");
    }
    }
    if (count == 16)
    {
      //add all the multip[lied numbers together and add then to the rest of the numbers
    for (int i = 0; i < count; i++)
  {
    temp = (number[i] % 10) + (number[i] / 10 % 10);
    v = v + temp;
   }
   //VISA check
      if (originalnumber[15] == 4 && v % 10 == 0)
    {
    printf("VISA\n");
    }
    //MASTERCARD chekc
  else if (originalnumber[15] == 5 && v % 10 == 0 && (originalnumber[14] == 1 || originalnumber[14] == 2 || originalnumber[14] == 3
  || originalnumber[14] == 4 || originalnumber[14] == 5))
        {
            printf("MASTERCARD\n");
        }
else
{
printf("INVALID\n");
}
}
}